src/ApiClient.php

```diff
@@ -292,18 +386,80 @@ public function getOptionChain(string $symbol, ?\DateTimeInterface $expiryDate =
         $qs = $this->getRandomQueryServer();
 
         // Initialize session cookies
-        $cookieJar = $this->getCookies();
+        $this->cookieJar = $this->getCookies();
 
         // Get crumb value
-        $crumb = $this->getCrumb($qs, $cookieJar);
+        $crumb = $this->getCrumb($qs);
 
         // Fetch options
         $url = 'https://query'.$qs.'.finance.yahoo.com/v7/finance/options/'.$symbol.'?crumb='.$crumb;
         if ($expiryDate) {
             $url .= '&date='.(string) $expiryDate->getTimestamp();
         }
-        $responseBody = (string) $this->client->request('GET', $url, ['cookies' => $cookieJar, 'headers' => $this->getHeaders()])->getBody();
+        try {
+            $responseBody = (string) $this->client->request('GET', $url, ['cookies' => $this->getCookies()])->getBody();
+
+            return $this->resultDecoder->transformOptionChains($responseBody);
+        } catch (\GuzzleHttp\Exception\ClientException $e) {
```

Comment:
As mentioned before, the `ClientException` isn't thrown for 401s, when you have `'http_errors' => false` set.